google/cloud/aiplatform/experimental/vertex_model/serializers/pandas.py

```diff
@@ -45,8 +48,10 @@ def _serialize_dataframe(artifact_uri: str, obj: pd.DataFrame, temp_dir: str) ->
 
     # Designate csv path and write the pandas DataFrame to the path
     # Convention: file name is my_training_dataset, my_test_dataset, etc.
-    path_to_csv = pathlib.Path(temp_dir) / ("my_dataset.csv")
-    obj.to_csv(path_to_csv)
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
The copying to GCS should also be within the block for this context manager so we do not lose the directory.